diff -urN ../apriltag-2015-03-18/apriltags.pc.in ./apriltags.pc.in
--- ../apriltag-2015-03-18/apriltags.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ./apriltags.pc.in	2015-04-30 17:16:00.383301282 +0200
@@ -0,0 +1,12 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: @TARGET_NAME@
+Description: @PROJECT_DESCRIPTION@
+Version: @PROJECT_VERSION@
+Requires: @PKGCONFIG_REQUIRES@
+Libs: -L${libdir} -l@TARGET_NAME@ @PKGCONFIG_LIBS@
+Cflags: -I${includedir} @PKGCONFIG_CFLAGS@
+
diff -urN ../apriltag-2015-03-18/CMakeLists.txt ./CMakeLists.txt
--- ../apriltag-2015-03-18/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ ./CMakeLists.txt	2015-04-30 17:41:36.927329329 +0200
@@ -0,0 +1,22 @@
+# CMakeLists.txt has to be located in the project folder and cmake has to be
+# executed from 'project/build' with 'cmake ../'.
+cmake_minimum_required(VERSION 2.6)
+find_package(Rock)
+rock_init(apriltags 0.1)
+
+set(CMAKE_C_FLAGS "-std=gnu99 -Wall")
+include_directories(common)
+
+FILE(GLOB all_sources RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" *.c common/*.c)
+FILE(GLOB all_headers RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" *.h common/*.h)
+rock_library(apriltags
+    SOURCES ${all_sources}
+    HEADERS ${all_headers}
+    DEPS_CMAKE Threads
+)
+
+rock_testsuite(apriltags_demo 
+   apriltag_demo.c
+   DEPS apriltags)
+
+target_link_libraries(apriltags pthread m)
diff -urN ../apriltag-2015-03-18/manifest.xml ./manifest.xml
--- ../apriltag-2015-03-18/manifest.xml	1970-01-01 01:00:00.000000000 +0100
+++ ./manifest.xml	2015-04-30 17:20:35.639306306 +0200
@@ -0,0 +1,12 @@
+<package>
+  <description brief="Apriltags Marker Detection, source file can be find in http://april.eecs.umich.edu/wiki/index.php/AprilTags">
+  
+  </description>
+  <author>April Robotics Laboratory, Diego Cesar/diego.cesar@dfki.de</author>
+  <maintainer>Diego Cesar/diego.cesar@dfki.de</maintainer>
+  <license></license>
+  <url>http://</url>
+  <logo>http://</logo>
+  <depend package="base/cmake" />
+  <depend package="base/types" />
+</package>
