diff --git a/autogen.sh b/autogen.sh
index 19384c2..471b4ed 100755
--- autogen.sh
+++ autogen.sh
@@ -27,13 +27,3 @@ intltoolize --force --copy --automake || exit 1
 gtkdocize --copy || exit 1
 autoreconf --verbose --force --install || exit 1
 
-cd "$olddir"
-if [ "$NOCONFIGURE" = "" ]; then
-        $srcdir/configure "$@" || exit 1
-
-        if [ "$1" = "--help" ]; then exit 0 else
-                echo "Now type 'make' to compile $PKG_NAME" || exit 1
-        fi
-else
-        echo "Skipping configure process."
-fi
