--- ../../../Aruco/aruco-1.2.5/utils/aruco_create_board.cpp	2012-12-10 09:29:39.000000000 +0100
+++ ./utils/aruco_create_board.cpp	2015-01-16 11:31:41.206313177 +0100
@@ -27,6 +27,7 @@
 ********************************/
 #include "board.h"
 #include <cstdio>
+#include <sstream>
 #include <opencv2/highgui/highgui.hpp>
 #include "arucofidmarkers.h"
 using namespace std;
@@ -49,7 +50,11 @@
 	int typeBoard=0;
         if (argc>=5) pixSize=atoi(argv[4]);
         if (argc>=6) typeBoard=atoi(argv[5]);
-        if (argc>=7) interMarkerDistance=atoi(argv[6]);
+        if (argc>=7) {
+            std::stringstream ss;
+            ss << argv[6];
+	    ss >> interMarkerDistance;
+        }
         aruco::BoardConfiguration BInfo;
         Mat BoardImage;
         if (typeBoard==0)
