--- CMakeLists.txt
+++ CMakeLists.txt
@@ -189,7 +189,7 @@ if (IOS)
   update_cache_variable(BUILD_EXAMPLES OFF)
   message(STATUS "Building for iOS, will not build examples.")
 
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fobjc-abi-version=2 -fobjc-arc -isysroot ${CMAKE_OSX_SYSROOT}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fobjc-abi-version=2 -fobjc-arc -fPIC -isysroot ${CMAKE_OSX_SYSROOT}")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fobjc-abi-version=2 -fobjc-arc -isysroot ${CMAKE_OSX_SYSROOT}")
 
    if (${IOS_PLATFORM} STREQUAL "SIMULATOR" OR
@@ -674,7 +674,7 @@ endif (MSVC)
 if (UNIX)
   # GCC is not strict enough by default, so enable most of the warnings.
   set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -Werror=all -Werror=extra -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers")
+      "${CMAKE_CXX_FLAGS} -Werror=all -Werror=extra -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-missing-field-initializers -fPIC")
 endif (UNIX)
 
 # Use a larger inlining threshold for Clang, since it hobbles Eigen,
