diff -urN ../factplusplus/Makefile ./Makefile
--- ../factplusplus/Makefile	2015-10-09 17:45:01.059568448 +0200
+++ ./Makefile	2015-10-09 17:41:36.815564779 +0200
@@ -3,7 +3,7 @@
 #
 
 # -- DO NOT CHANGE THE REST OF FILE --
-SUBDIRS = Kernel FaCT++ FaCT++.JNI FaCT++.C
+SUBDIRS = Kernel FaCT++ FaCT++.C
 #SUBDIRS = Kernel Tests
 
 include Makefile.include
@@ -12,13 +12,13 @@
 
 .PHONY: kernel
 kernel:
-	make -C Kernel
+	make -C Kernel -j1
 
 .PHONY: fpp_lisp
 fpp_lisp: kernel
-	make -C FaCT++
+	make -C FaCT++ -j1
 
 .PHONY: fpp_jni
 fpp_jni: kernel
-	make -C FaCT++.JNI
+	make -C FaCT++.JNI -j1
 
diff -urN ../factplusplus/Makefile.include ./Makefile.include
--- ../factplusplus/Makefile.include	2015-10-09 17:45:01.059568448 +0200
+++ ./Makefile.include	2015-10-09 17:39:47.163562809 +0200
@@ -21,7 +21,7 @@
 
 # define architecture; it will affect file extentions, dynamic flags.
 # now LINUX, WINDOWS(untested), MACOSX are supported
-OS = MACOSX
+OS = LINUX
 
 # ALL_DEF is an external defines
 GENERAL_DEFINES = $(GCC_OPT_OPT) $(ALL_DEF)
