--- CMakeLists.txt.orig	2021-09-10 11:30:52.689973400 +0000
+++ CMakeLists.txt	2021-09-10 11:31:50.583087612 +0000
@@ -488,6 +488,15 @@
 # Set up the top-level include directories
 include_directories(${g2o_SOURCE_DIR} ${PROJECT_BINARY_DIR})
 
+
+# Export libraries via pkgconfig
+SET(PKG_CONFIG_EXPORT_LIBS "-lg2o_cli -lg2o_csparse_extension -lg2o_hierarchical -lg2o_solver_dense -lg2o_solver_pcg -lg2o_stuff -lg2o_types_slam3d -lg2o_core -lg2o_ext_freeglut_minimal -lg2o_opengl_helper -lg2o_solver_eigen -lg2o_solver_structure_only -lg2o_types_slam2d")
+
+configure_file(g2o.pc.in g2o.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/g2o.pc DESTINATION lib/pkgconfig)
+
+
+
 # Generate config.h
 set(G2O_OPENGL_FOUND ${OPENGL_FOUND})
 set(G2O_HAVE_CHOLMOD ${CHOLMOD_FOUND})
diff -Naur g2o.pc.in g2o.pc.in
--- g2o.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ g2o.pc.in	2012-10-28 12:21:09.129485817 +0100
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: g2o 
+Description: slam backend
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} @PKG_CONFIG_EXPORT_LIBS@
+Cflags: -I${includedir}
+
