diff -Naur g2o/core/graph_optimizer_sparse.h g2o/core/graph_optimizer_sparse.h
--- g2o/core/graph_optimizer_sparse.h	2013-05-02 15:33:45.282886579 +0200
+++ g2o/core/graph_optimizer_sparse.h 	2013-05-02 15:33:35.222886714 +0200
@@ -178,0 +179,5 @@
+    void push();
+    void pop();
+    void discardTop();
+    bool removeLastEdge();
+
diff -Naur g2o/core/graph_optimizer_sparse.cpp g2o/core/graph_optimizer_sparse.cpp
--- g2o/core/graph_optimizer_sparse.cpp	2013-05-02 15:33:45.278886579 +0200
+++ g2o/core/graph_optimizer_sparse.cpp 	2013-05-02 15:33:28.774886803 +0200
@@ -613,0 +614,5 @@
+  void SparseOptimizer::push()
+  {
+    push(_activeVertices);
+  }
+
@@ -619,0 +625,5 @@
+  void SparseOptimizer::pop()
+  {
+    pop(_activeVertices);
+  }
+
@@ -647,0 +658,5 @@
+  void SparseOptimizer::discardTop()
+  { 
+    discardTop(_activeVertices);
+  }
+
@@ -694,0 +710,6 @@
+  }
+
+  bool SparseOptimizer::removeLastEdge()
+  {
+    Edge *e = _activeEdges[_activeEdges.size() - 1];
+    return HyperGraph::removeEdge(e);
