diff -Naur CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt	2012-10-28 12:16:38.761489438 +0100
+++ CMakeLists.txt	2012-10-28 12:20:57.289485976 +0100
@@ -153,0 +153,5 @@
+
+configure_file(g2o.pc.in g2o.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/g2o.pc DESTINATION lib/pkgconfig)
+
+
diff -Naur g2o.pc.in g2o.pc.in
--- g2o.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ g2o.pc.in	2012-10-28 12:21:09.129485817 +0100
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: g2o 
+Description: slam backend
+Version: @PROJECT_VERSION@
+Libs: -L${libdir} -lg2o_cli -lg2o_core -lg2o_csparse_extension -lg2o_interface -lg2o_simulator -lg2o_solver_csparse -lg2o_solver_dense -lg2o_solver_eigen -lg2o_solver_pcg -lg2o_solver_slam2d_linear -lg2o_solver_structure_only -lg2o_stuff -lg2o_types_data -lg2o_types_icp -lg2o_types_sba -lg2o_types_sclam2d -lg2o_types_sim3 -lg2o_types_slam2d -lg2o_types_slam3d
+Cflags: -I${includedir}
+
diff -Naur manifest.xml manifest.xml
--- manifest.xml  1970-01-01 01:00:00.000000000 +0100
+++ manifest.xml  2012-10-28 12:21:09.129485817 +0100
@@ -0,0 +1,15 @@
+<package>
+    <description brief="g2o: A General Framework for Graph Optimization">
+    g2o is an open-source C++ framework for optimizing graph-based nonlinear
+    error functions.  g2o has been designed to be easily
+    extensible to a wide range of problems and a new problem typically
+    can be specified in a few lines of code.  The current implementation
+    provides solutions to several variants of SLAM and BA.
+    </description>
+    <author>Giorgio Grisetti; Rainer Kuemmerle; Hauke Strasdat; Kurt Konolige; Wolfram Burgard </author>
+    <copyright>
+       Copyright and V.i.S.d.P.: Giorgio Grisetti; Rainer Kuemmerle; Hauke Strasdat; Kurt Konolige; Wolfram Burgard
+    </copyright>
+    <license>LGPL v.3</license>
+    <url>http://www.openslam.org/g2o.html</url>
+</package>
