--- CMakeLists.txt
+++ CMakeLists.txt
@@ -336,6 +336,10 @@ if(GTSAM_ENABLE_CONSISTENCY_CHECKS)
   add_definitions(-DGTSAM_EXTRA_CONSISTENCY_CHECKS)
 endif()
 
+if(NOT WIN32)
+    option(GTSAM_BUILD_PKGCONFIG "Build pkg-config .pc file for GTSAM" ON)
+endif(NOT WIN32)
+
 ###############################################################################
 # Add components
 
--- gtsam/CMakeLists.txt
+++ gtsam/CMakeLists.txt
@@ -158,3 +158,46 @@ if (GTSAM_INSTALL_MATLAB_TOOLBOX)
     # Wrap
     wrap_and_install_library(../gtsam.h "${GTSAM_ADDITIONAL_LIBRARIES}" "" "${mexFlags}")
 endif ()
+
+macro(libraries_for_pkgconfig VARNAME)
+    foreach(__lib ${ARGN})
+        string(STRIP __lib ${__lib})
+        string(SUBSTRING ${__lib} 0 1 __lib_is_absolute)
+        if (__lib_is_absolute STREQUAL "/")
+            get_filename_component(__lib_path ${__lib} PATH)
+            get_filename_component(__lib_name ${__lib} NAME_WE)
+            string(REGEX REPLACE "^lib" "" __lib_name "${__lib_name}")
+            set(${VARNAME} "${${VARNAME}} -L${__lib_path} -l${__lib_name}")
+        else()
+            set(${VARNAME} "${${VARNAME}} ${__lib}")
+        endif()
+    endforeach()
+endmacro()
+
+# Generate pkgconfig pc file
+if(GTSAM_BUILD_PKGCONFIG)
+    if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/gtsam.pc.in)
+        message("-- Found: ${CMAKE_CURRENT_SOURCE_DIR}/gtsam.pc.in")
+
+        # Set the target name
+        set(TARGET_NAME gtsam)
+
+        # Dependency for GTSAM
+        libraries_for_pkgconfig(${TARGET_NAME}_PKGCONFIG_LIBS
+            ${GTSAM_BOOST_LIBRARIES} ${GTSAM_ADDITIONAL_LIBRARIES}
+            ${LAPACK_LIBRARIES})
+
+        set(${TARGET_NAME}_PKGCONFIG_CFLAGS "${${TARGET_NAME}_PKGCONFIG_CFLAGS} ${BOOST_CFLAGS_OTHER}")
+
+        set(PKGCONFIG_REQUIRES eigen3)
+        set(PKGCONFIG_CFLAGS ${${TARGET_NAME}_PKGCONFIG_CFLAGS})
+        set(PKGCONFIG_LIBS ${${TARGET_NAME}_PKGCONFIG_LIBS})
+
+        configure_file(gtsam.pc.in gtsam.pc @ONLY)
+        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/gtsam.pc
+            DESTINATION lib/pkgconfig)
+    else()
+        message("-- pkg-config: ${CMAKE_CURRENT_SOURCE_DIR}/gtsam.pc.in is not available for configuration")
+    endif()
+
+endif(GTSAM_BUILD_PKGCONFIG)
--- /dev/null
+++ ./gtsam/gtsam.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+
+Name: @TARGET_NAME@
+Description: @TARGET_NAME@
+Version: @GTSAM_VERSION_STRING@
+Requires: @PKGCONFIG_REQUIRES@
+Libs: -L${libdir} -l@TARGET_NAME@ @PKGCONFIG_LIBS@
+Cflags: -I${includedir} @PKGCONFIG_CFLAGS@
