Index: aislib/graph_optimizer_hogman/graph_optimizer_chol.hpp
===================================================================
--- aislib/graph_optimizer_hogman/graph_optimizer_chol.hpp	(Revision 21)
+++ aislib/graph_optimizer_hogman/graph_optimizer_chol.hpp	(Arbeitskopie)
@@ -53,7 +53,7 @@
       root=_MY_CAST_<typename PG::Vertex*>(this->vertices().begin()->second);
     if (this->verbose())
       cerr << "# root id " << root->id() << endl;
-    bool initFromObservations = _guessOnEdges;
+    bool initFromObservations = this->guessOnEdges();
     optimizeSubset(root, vset, iterations, 0., initFromObservations);
     return iterations;
   }
@@ -281,7 +281,7 @@
 
     if (eset.empty()){
       typename PG::Edge* e = PG::addEdge(from, to, mean, information);
-      if (_guessOnEdges && to->edges().size()==1 && ! to->fixed()){
+      if (this->guessOnEdges() && to->edges().size()==1 && ! to->fixed()){
 	to->transformation=from->transformation*mean;
       }
       return e;
