--- CMakeLists.txt.old	2013-08-20 15:38:10.202916901 +0200
+++ CMakeLists.txt	2013-08-20 15:39:09.062917030 +0200
@@ -1,5 +1,8 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+find_package(Rock)
+rock_init(mtk 1.0)
 
+rock_find_pkgconfig(Eigen3 eigen3 REQUIRED)
 
 IF(NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE RELEASE CACHE STRING
@@ -19,11 +22,11 @@
 SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules)
 
 FIND_PACKAGE(CXSparse REQUIRED) 
-FIND_PACKAGE(Eigen2)
+FIND_PACKAGE(LAPACK)
 FIND_PACKAGE(Boost 1.36.0 REQUIRED COMPONENTS program_options unit_test_framework)
 
 INCLUDE_DIRECTORIES(${CXSPARSE_INCLUDE_DIR} )
-INCLUDE_DIRECTORIES(${EIGEN2_INCLUDE_DIR} )
+INCLUDE_DIRECTORIES(${EIGEN3_INCLUDE_DIR} )
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR} )
 
 INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIRS})
@@ -43,3 +46,41 @@
 	ADD_SUBDIRECTORY (example)
 	ADD_SUBDIRECTORY (unit_test)
 ENDIF()
+
+INSTALL(FILES
+    mtk/startIdx.hpp
+    mtk/build_manifold.hpp
+    mtk/mean_and_covar.hpp
+    DESTINATION include/mtk)
+
+INSTALL(FILES
+    mtk/src/eigen.hpp
+    mtk/src/mtkmath.hpp
+    mtk/src/SubManifold.hpp
+    mtk/src/vectview.hpp
+    DESTINATION include/mtk/src)
+
+INSTALL(FILES
+    mtk/types/pose.hpp
+    mtk/types/S2.hpp
+    mtk/types/SOn.hpp
+    mtk/types/vect.hpp
+    DESTINATION include/mtk/types)
+
+INSTALL(FILES
+    ukfom/mtkwrap.hpp
+    ukfom/util.hpp
+    ukfom/ukf.hpp
+    DESTINATION include/ukfom)
+
+INSTALL(FILES
+    ukfom/lapack/cholesky.hpp
+    ukfom/lapack/invert.hpp
+    ukfom/lapack/lapack.h
+    DESTINATION include/ukfom/lapack)
+
+INSTALL(FILES
+    ukfom/traits/dof.hpp
+    DESTINATION include/ukfom/traits)
+
+
