--- a/orocos_kdl/CMakeLists.txt
+++ b/orocos_kdl/CMakeLists.txt
@@ -45,12 +45,13 @@ ENDIF ( NOT CMAKE_BUILD_TYPE )

 SET( KDL_CFLAGS "")

-find_package(Eigen3 QUIET)
-if(NOT EIGEN3_FOUND)
-  include(${PROJ_SOURCE_DIR}/cmake/FindEigen3.cmake)
+rock_find_pkgconfig(Eigen 3.0 QUIET)
+if(NOT Eigen_FOUND)
+   include(${PROJ_SOURCE_DIR}/cmake/FindEigen3.cmake)
+   set(EIGEN_INCLUDE_DIR ${EIGEN3_INCLUDE_DIR})
 endif()
-include_directories(${EIGEN3_INCLUDE_DIR})
-SET(KDL_CFLAGS "${KDL_CFLAGS} -I${EIGEN3_INCLUDE_DIR}")
+include_directories(${EIGEN_INCLUDE_DIR})
+SET(KDL_CFLAGS "${KDL_CFLAGS} -I${EIGEN_INCLUDE_DIR}")
