diff -urN kdl/orocos_kdl/CMakeLists.txt kdl_new/orocos_kdl/CMakeLists.txt
--- kdl/orocos_kdl/CMakeLists.txt	2013-02-08 14:25:12.688429140 +0100
+++ kdl_new/orocos_kdl/CMakeLists.txt	2013-02-08 14:19:01.688445787 +0100
@@ -43,7 +43,7 @@
 
 SET( KDL_CFLAGS "" CACHE INTERNAL "")
 
-find_package(Eigen 3.0 QUIET)
+rock_find_pkgconfig(Eigen3 eigen3 REQUIRED)
 if(NOT Eigen_DIR)
 include(${PROJ_SOURCE_DIR}/config/FindEigen3.cmake)
 include_DIRECTORIES(${EIGEN3_INCLUDE_DIR})
