diff -urN ../snap/Makefile ./Makefile
--- ../snap/Makefile	2013-12-05 17:09:31.455157716 +0100
+++ ./Makefile	2013-12-05 17:08:44.575158163 +0100
@@ -4,10 +4,14 @@
 
 all: MakeAll
 
+lib: MakeAll
+	$(MAKE) -C snap-core lib
+
 MakeAll:
 	$(MAKE) -C snap-core
 	$(MAKE) -C examples
 
+
 clean:
 	$(MAKE) clean -C snap-core
 	$(MAKE) clean -C examples
--- snap-core/Makefile	2016-08-05 15:56:25.164361591 +0200
+++ ../snap2/snap-core/Makefile	2016-08-05 15:52:39.627243213 +0200
@@ -13,11 +13,11 @@
 	$(CC) -c $(CXXFLAGS) Snap.cpp -I$(CGLIB)
 
 lib: Snap.o
-	rm -f libsnap.a
-	ar -cvq libsnap.a Snap.o
+	rm -f ../build/libsnap.a
+	ar -cvq ../build/libsnap.a Snap.o
 
 libinstall: lib
-	cp libsnap.a /usr/local/lib/libsnap.a
+	cp ../build/libsnap.a /usr/local/lib/libsnap.a
 
 clean:
 	rm -f *.o *.a testSnap testSnap.exe *.plt *.tab *.png
