diff -ruN ../libviso2/CMakeLists.txt ./CMakeLists.txt
--- ../libviso2/CMakeLists.txt	2013-09-09 10:22:40.000000000 +0200
+++ ./CMakeLists.txt	2014-01-21 18:23:48.109233575 +0100
@@ -1,23 +1,10 @@
 # project
 cmake_minimum_required (VERSION 2.6)
-project (libviso2)
-
-# directories
-set (LIBVISO2_SRC_DIR src)
-
-# include directory
-include_directories("${LIBVISO2_SRC_DIR}")
-
-# use sse3 instruction set
-SET(CMAKE_CXX_FLAGS "-msse3")
-
-# sources
-FILE(GLOB LIBVISO2_SRC_FILES "src/*.cpp")
+find_package(Rock)
+rock_init(viso2 0.1)
+rock_standard_layout()
 
 # make release version
 set(CMAKE_BUILD_TYPE Release)
 
-# demo program
-add_executable(viso2 ${LIBVISO2_SRC_FILES})
-target_link_libraries (viso2 png)
 
diff -ruN ../libviso2/src/CMakeLists.txt ./src/CMakeLists.txt
--- ../libviso2/src/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ ./src/CMakeLists.txt	2014-01-21 18:29:57.865234049 +0100
@@ -0,0 +1,35 @@
+# use sse3 instruction set
+SET(CMAKE_CXX_FLAGS "-msse3")
+
+set(VISO2_SRCS
+    filter.cpp
+    matcher.cpp
+    matrix.cpp
+    reconstruction.cpp
+    triangle.cpp
+    viso.cpp
+    viso_mono.cpp
+    viso_stereo.cpp
+    )
+
+set (VISO2_HDRS
+    filter.h
+    matcher.h
+    matrix.h
+    reconstruction.h
+    timer.h
+    triangle.h
+    viso.h
+    viso_mono.h
+    viso_stereo.h
+    )
+
+
+rock_library(viso2
+    ${VISO2_SRCS}
+    HEADERS ${VISO2_HDRS})
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+add_executable(viso2_bin demo.cpp)
+target_link_libraries (viso2_bin viso2 png)
+
diff -ruN ../libviso2/src/viso2.pc.in ./src/viso2.pc.in
--- ../libviso2/src/viso2.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ ./src/viso2.pc.in	2014-01-21 17:38:16.693230067 +0100
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/lib
+includedir=${prefix}/include
+ 
+Name: @TARGET_NAME@
+Description: @PROJECT_DESCRIPTION@
+Version: @PROJECT_VERSION@
+Requires: @DEPS_PKGCONFIG@
+Libs: -L${libdir} -l@TARGET_NAME@
+Cflags: -I${includedir} -msse2
